whitenoise/middleware.py

```diff
@@ -168,3 +181,45 @@ def get_static_url(self, name):
             return decode_if_byte_string(staticfiles_storage.url(name))
         except ValueError:
             return None
+
+        
+
+        
+FILE_FORMATS_WITH_WEBP_SUBSTITION = ['.jpg', '.png', '.jpeg']
+
+class WebPWhiteNoiseMiddleware(WhiteNoiseMiddleware):
```

Comment:
`webp` stuff would best be done in the primary middleware class, rather than creating a whole new one. Then, allow for a django `settings.py` value of something like `WHITENOISE_WEBP_SUBSTITUTION: bool = ...` to enable or disable this behavior.